libraries/DataFlash/LogFile.cpp

```diff
@@ -1545,71 +1545,54 @@ void DataFlash_Class::Log_Write_AttitudeView(AP_AHRS_View &ahrs, const Vector3f
     WriteBlock(&pkt, sizeof(pkt));
 }
 
-// Write an Current data packet
-void DataFlash_Class::Log_Write_Current(const AP_BattMonitor &battery)
+void DataFlash_Class::Log_Write_Current_instance(const AP_BattMonitor &battery,
+                                                 const uint64_t time_us,
+                                                 const uint8_t battery_instance,
+                                                 const enum LogMessages type)
 {
-    if (battery.num_instances() >= 1) {
-        float temp;
-        bool has_temp = battery.get_temperature(temp, 0);
-        struct log_Current pkt = {
-            LOG_PACKET_HEADER_INIT(LOG_CURRENT_MSG),
+    float temp;
+    bool has_temp = battery.get_temperature(temp, battery_instance);
+    struct log_Current pkt = {
+        LOG_PACKET_HEADER_INIT(type),
+        time_us             : AP_HAL::micros64(),
+        voltage             : battery.voltage(battery_instance),
+        voltage_resting     : battery.voltage_resting_estimate(battery_instance),
+        current_amps        : battery.current_amps(battery_instance),
+        current_total       : battery.current_total_mah(battery_instance),
+        temperature         : (int16_t)(has_temp ? (temp * 100) : battery_instance),
```

Comment:
This `battery_instance` is wrong, it should stay a 0. That search and replace failed this time 😜